question/bank/importquestions/import.php

```diff
@@ -0,0 +1,148 @@
+    }
+    $contexts->require_one_edit_tab_cap($edittab);
+}
+
+$PAGE->set_url($thispageurl);
+
+$import_form = new question_import_form($thispageurl, array('contexts'=>$contexts->having_one_edit_tab_cap('import'),
+                                                    'defaultcategory'=>$pagevars['cat']));
+
+if ($import_form->is_cancelled()){
+    redirect($thispageurl);
+}
+//==========
+// PAGE HEADER
+//==========
+$PAGE->set_title($txt->importquestions);
+$PAGE->set_heading($COURSE->fullname);
+echo $OUTPUT->header();
+
+// Print horizontal nav if needed.
+$renderer = $PAGE->get_renderer('core_question', 'bank');
+echo $renderer->extra_horizontal_navigation();
+
+// file upload form sumitted
+if ($form = $import_form->get_data()) {
+
+    // file checks out ok
+    $fileisgood = false;
+
+    // work out if this is an uploaded file
+    // or one from the filesarea.
+    $realfilename = $import_form->get_new_filename('newfile');
+    $importfile = make_request_directory() . "/{$realfilename}";
+    if (!$result = $import_form->save_file('newfile', $importfile, true)) {
+        throw new moodle_exception('uploadproblem');
+    }
+
+    $formatfile = $CFG->dirroot . '/question/format/' . $form->format . '/format.php';
+    if (!is_readable($formatfile)) {
+        throw new moodle_exception('formatnotfound', 'question', '', $form->format);
+    }
+
+    require_once($formatfile);
+
+    $classname = 'qformat_' . $form->format;
+    $qformat = new $classname();
+
+    // load data into class
+    $qformat->setCategory($category);
+    $qformat->setContexts($contexts->having_one_edit_tab_cap('import'));
+    $qformat->setCourse($COURSE);
+    $qformat->setFilename($importfile);
+    $qformat->setRealfilename($realfilename);
+    $qformat->setMatchgrades($form->matchgrades);
+    $qformat->setCatfromfile(!empty($form->catfromfile));
+    $qformat->setContextfromfile(!empty($form->contextfromfile));
+    $qformat->setStoponerror($form->stoponerror);
+
+    // Do anything before that we need to
+    if (!$qformat->importpreprocess()) {
+        print_error('cannotimport', '', $thispageurl->out());
+    }
+
+    // Process the uploaded file
+    if (!$qformat->importprocess()) {
+        print_error('cannotimport', '', $thispageurl->out());
+    }
+
+    // In case anything needs to be done after
+    if (!$qformat->importpostprocess()) {
+        print_error('cannotimport', '', $thispageurl->out());
+    }
+
+    // Log the import into this category.
+    $eventparams = [
+            'contextid' => $qformat->category->contextid,
+            'other' => ['format' => $form->format, 'categoryid' => $qformat->category->id],
+    ];
+    $event = \core\event\questions_imported::create($eventparams);
+    $event->trigger();
+
+    $params = $thispageurl->params() + array(
+        'category' => $qformat->category->id . ',' . $qformat->category->contextid);
+    echo $OUTPUT->continue_button(new moodle_url('edit.php', $params));
```

Comment:
Probably, as we want that everything works while we are adding new plugins th continue button should be redirect to the current edit file:

`echo $OUTPUT->continue_button(new moodle_url('/question/edit.php', $params));`

Reply:
Continue button redirection set to current `/question/edit.php`